src/format.rs

```diff
@@ -43,23 +43,16 @@ impl fmt::Display for FormattedDuration {
 
 impl fmt::Display for HumanDuration {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let t = self.0.as_secs();
+        let t = self.0.as_millis();
         let alt = f.alternate();
         macro_rules! try_unit {
             ($secs:expr, $sg:expr, $pl:expr, $s:expr) => {
-                let cnt = t / $secs;
-                if cnt == 1 {
+                let cnt = t / (500 * $secs);
```

Comment:
This is looking really good, thanks for the extensive tests!

Just one more nit: can you add a few lines of comments here that explain the strategy here? In particular, why we are using ms instead of seconds and are operating in half-seconds here.

Reply:
Thank you for your comment, but I managed to simplify this logic and I don't need milliseconds nor half units here any more.